spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/SimpleJobService.java

```diff
@@ -256,14 +257,19 @@ public int countJobs() {
 	@Override
 	public int stopAll() {
 		Collection<JobExecution> result = jobExecutionDao.getRunningJobExecutions();
-		Collection<String> jsrJobNames = getJsrJobNames();
+		Collection<String> jsrJobNames = getJobNames();
 
 		for (JobExecution jobExecution : result) {
-			if (jsrJobOperator != null && jsrJobNames.contains(jobExecution.getJobInstance().getJobName())) {
-				jsrJobOperator.stop(jobExecution.getId());
-			} else {
-				jobExecution.stop();
-				jobRepository.update(jobExecution);
+			if (jsrJobNames.contains(jobExecution.getJobInstance().getJobName())) {
+                try {
+                    jobOperator.stop(jobExecution.getId());
+                } catch (NoSuchJobExecutionException e) {
```

Comment:
The top 2 catch can be consolidated via an `|` in the catch clause.

Reply:
Changed it to a catch for `Exception`